source/Halibut.Tests/Support/TestCases/ClientAndServiceTestCasesBuilder.cs

```diff
@@ -52,26 +59,30 @@ public IEnumerable<ClientAndServiceTestCase> Build()
                             
                             if (!forceClientProxyTypes.Any())
                             {
-                                yield return new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion, null, serviceAsyncHalibutFeatureTestCase);
+                                cases.Add(new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion, null, serviceAsyncHalibutFeatureTestCase));
                             }
                             else
                             {
                                 if (clientServiceTestVersion.IsPreviousClient())
                                 {
-                                    yield return new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion, null, serviceAsyncHalibutFeatureTestCase);
+                                    cases.Add(new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion, null, serviceAsyncHalibutFeatureTestCase));
                                 }
                                 else
                                 {
                                     foreach (var forceClientProxyType in forceClientProxyTypes)
                                     {
-                                        yield return new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion, forceClientProxyType, serviceAsyncHalibutFeatureTestCase);
+                                        cases.Add(new ClientAndServiceTestCase(serviceConnectionType, networkConditionTestCase, recommendedIterations, clientServiceTestVersion, forceClientProxyType, serviceAsyncHalibutFeatureTestCase));
                                     }
                                 }
                             }
                         }
                     }
                 }
             }
+
+            var distinct = cases.Distinct().ToList();
```

Comment:
The logic above can generate duplication and I was seeing this locally. Running 1 test resulted in 2 results 😬